File: 0001TwoSum/main.py
def twoSum(nums, target):
    size = len(nums)
    answer = ()

    results = [(num, i) for i, num in enumerate(nums)]
    results.sort(key= lambda x: x[0])
    print(results)

    i = 0
    j = size - 1
    while i < j:
        sum = results[i][0] + results[j][0]
        if sum > target:
            j = j - 1
        elif sum < target:
            i = i + 1
        else:
            answer = (results[i][1], results[j][1])
            break
    
    return answer

if __name__ == "__main__":
    print(twoSum([3,2,3], 6))

File: 0001TwoSum/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const size = nums.length;
    const answer = [0, 0];

    const result = nums.map((num, i) => [num, i]);
    result.sort((a, b) => a[0] - b[0]);

    i = 0, j = size - 1;
    while(i < j) {
        const sum = result[i][0] + result[j][0];
        if(sum > target) {
            j--;
        } else if(sum < target) {
            i++;
        } else  {
            answer[0] = result[i][1];
            answer[1] = result[j][1];
            break;
        }
    }

    return answer;
};

console.log(twoSum([2,7,11,15], 9));